core/clingutils/CMakeLists.txt

```diff
@@ -45,12 +50,27 @@ foreach(dict ${stldicts})
   string(REPLACE "2" "" header ${dict})
   string(REPLACE "complex" "root_std_complex.h" header ${header})
   string(REPLACE "multi" "" header ${header})
+
+  # Manually search for the header in our STL folders because the ROOT macros
+  # could pick up other files like /bin/map due to their extensive search paths.
+  foreach(path ${cxx_path})
+    find_file(abs_cling_header name ${header} HINTS ${path} NO_DEFAULT_PATH)
```

Comment:
Here it would be better to use `find_path()` instead of `find_file()` to restrict the kinds of places where lookups happen.

Reply:
Also, instead of doing a manual `foreach()` it is probably better to just pass the list to `find_path()` command via the `PATHS` option. https://cmake.org/cmake/help/latest/command/find_path.html

Reply:
`find_path` is exactly the same as `find_file` in every aspect beside that it looks for a directory. And we look for a header *file*, so `find_file` it is :)

I replaced the for loop, that was some `message()` driven development artifact.